libvast/src/http_api.cpp

```diff
@@ -54,29 +54,29 @@ auto parse_skeleton(simdjson::ondemand::value value, size_t depth = 0)
 
 auto http_parameter_map::from_json(std::string_view json)
   -> caf::expected<http_parameter_map> {
-  http_parameter_map result;
-  if (json.empty())
+  try {
+    http_parameter_map result;
+    if (json.empty())
+      return result;
+    auto padded_string = simdjson::padded_string{json};
+    auto parser = simdjson::ondemand::parser{};
+    auto body = simdjson::ondemand::object{};
+    auto doc = parser.iterate(padded_string).value();
+    if ([[maybe_unused]] auto error = doc.get_object().get(body))
+      return caf::make_error(ec::invalid_argument,
+                             "expected a top-level object");
```

Comment:
If we're switching to exceptions we can probably remove the use of the non-throwing API here as well and rely completely on them.